x-pack/plugins/security_solution/server/lib/detection_engine/signals/search_after_bulk_create.ts

```diff
@@ -208,6 +290,7 @@ export const searchAfterAndBulkCreate = async ({
         // we are guaranteed to have searchResult hits at this point
         // because we check before if the totalHits or
         // searchResult.hits.hits.length is 0
+        // call this function setSortIdOrExit()
         const lastSortId = searchResult.hits.hits[searchResult.hits.hits.length - 1].sort;
```

Comment:
Since `searchResultB` is pushed on to the end of `searchResult` now this sortId won't be correct. Setting the sortID immediately after calling `singleSearchAfter` will avoid this issue. We also only want to `break` if both the primary and backup searches have reached the end, i.e. `searchResult.hits.hits.length === 0`

Reply:
@marshallmain Can sort id's be used interchangeably between queries with different sort fields? If I sort by `event.ingested` in one search and then sort by `@timestamp` in the second search, can I loop back up to the first `singleSearchAfter` and use the sort ID from the `@timestamp` search as a part of the `search_after` for the `event.ingested` query? If that is possible, is that something we want to mix between the searches?

Reply:
We can't mix the sort IDs - the sort ID keeps track of the last document returned by the query. Since the primary and backup queries are querying different sets documents we'll get incorrect results if we attempt to use the sort ID from one query as part of the other one. 

e.g. if the last sort ID from the `@timestamp` query is `555` and the last sort ID from the `event.ingested` query is `100`, then if we were to use the last sort ID from `@timestamp` in the next `event.ingested` query we would miss all the documents from `100` to `555`. To make the consecutive searches work properly we need to use the sort ID from the previous `event.ingested` query in the next `event.ingested` query, and the previous `@timestamp` query in the next `@timestamp` query.

Reply:
Are you implying that there is mixing of the sortId's in the code as-is? I'm not sure I'm seeing that mixing..

Reply:
@dhurley14 This code is supposed to be retrieving the last `sortId` from the `timestampOverride` sorted set (used above, here: https://github.com/elastic/kibana/pull/86368/files#diff-70a938251572c8b374514dbb063ecc7599afbb8b74aec071023c54239087662dR90) ... the problem is we've merged the `@timestamp` sorted set in with the `timestampOverride` set. So your sortIds will be "mixed up" at this point, and unless the set was resorted (doesn't look like it), both `timestampOverride` and `@timestamp` sort IDs will be the same.

You'll feed the wrong `sortId` into the next query and miss anything between the end of your `timestampOverride` query and your `sortId`. I believe this is what @marshallmain was pointing out.

Reply:
Ah yeah I see the bug. Couldn't follow it directly from your comment. Good catch!

Reply:
To fix, you'll need to track both separately. Probably just calculate this `sortId` before the sets are merged, unless there is some other reason to avoid that. And note the comment above... the `searchResult.hits` length check isn't performed until after the sets are already merged.

Reply:
Ah yeah thanks @madirey :)

Reply:
Yeah @madirey 
https://github.com/elastic/kibana/pull/86368/files#diff-70a938251572c8b374514dbb063ecc7599afbb8b74aec071023c54239087662dR142
On this line the hits from `searchResultB` get pushed on to the end of `searchResult.hits.hits` - so here when we get the last element of `searchResult.hits.hits` it'll be from `searchResultB` instead of the original `searchResult`

Reply:
Sweet, sounds like you're on the right track. 👍